gcp/appengine/frontend_handlers.py

```diff
@@ -190,6 +191,39 @@ def add_source_info(bug, response):
   response['source_link'] = response['source']
 
 
+def add_related_aliases(bug: osv.Bug, response):
+  """Add links to other osv entries that's related through aliases"""
+  # Add links to other entries if they exist
+  aliases = {}
+  for alias in bug.aliases:
+    result = bug.get_by_id(alias)
+    aliases[alias] = {'exists': result, 'same_alias_entries': []}
+
+  # Add links to other entries that have the same alias or references this
+  if bug.aliases:
+    query = osv.Bug.query(osv.Bug.aliases.IN(bug.aliases + [bug.id()]))
+    for other in query:
+      if other.id() == bug.id():
+        continue
+      for other_alias in other.aliases:
+        if other_alias in aliases:
+          aliases[other_alias]['same_alias_entries'].append(other.id())
+    # Add links to other entries if they have this element as an alias
+    # This has to be a separate loop to avoid the new aliases entries interfering
+    for other in query:
```

Comment:
I think this actually runs the query twice. 

Could you explain what the interference here would be in more detail?

Reply:
Oh... in that case we might want to just put it in the same loop.

The main thing that this is avoiding is inconsistently showing the `(<same-alias>)` links in brackets for the newly added direct references. Because let's say a direct reference `x` is added on loop 4 / 10, if there was another entry that happened to have an alias of `x`, it will not show up if it has already been processed in an earlier loop. 

This is probably pretty rare though, so we can start by putting it into the same loop until it becomes a problem in the future. (We can also pull the entire query into a list in memory and loop twice that way?)

Reply:
Hmm.. not sure i fully follow. Could we not just set `exists` to be True in the direct reference case and not touch the existing `same_alias_entries` case?